### ISSUE TITLE
Better representation for caption sentences.

### ISSUE THREAD

**Issue by siyuzhou** [siyuzhou/AIWC_Commentator#1]
Right now, all possible comments are gathered to form a caption "list", in which each sentence is labeled by their index in the list. The ground truth labels are these integer indices, and a weighted cross entropy is used as training loss. This approach has a problem: expressive variation in a sentence that would provide the same meaning is now treated as different. Different sentences of the same meaning compete for the model's predictive power. An otherwise correct caption interchangeable with the ground truth semantically is considered a bad prediction. Further work can be done to find the connections between the comments using better representation.
